### DISCUSSION TITLE
How did you address the CC2531 stick

### DISCUSSION THREAD

**Discussion by 1newProject** [Koenkk/zigbee2mqtt#14211] [General]
Hello Zigbee2MQTT Team,
me and my team are studying at the TH Köln. We are interested in the topic of Zigbee and are currently testing out the limits of Philips Hue lamps as part of a Zigbee security analysis.
We have set up Zigbee2MQTT and can use it to control our lamps. It works very well. Unfortunately we can't send our own packets using Z2M, which is of course not the purpose of the software, however we need to do this for our project.
We have been trying to send our own packets for some time, but just can't get it to work. We would be very happy if you could help us how and where exactly in the code you did the addressing and sending the Zigbee packages to the CC2531 stick.
With kind regards
Christian

**Comment by Koenkk** [comment]
You can write a custom converter for your device which allows you to send whatever Zigbee command you want. https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions should help you getting started. (check the [advanced example `tzLocal`](https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/externalConvertersExample/freepad_ext.js)
